lms/djangoapps/verify_student/views.py

```diff
@@ -1047,3 +1058,142 @@ def reverification_window_expired(_request):
+        if checkpoint is None:
+            log.error("Checkpoint is not defined. Could not submit verification attempt for user %s",
+                      request.user.id)
+            context = {
+                'course_key': unicode(course_key),
+                "course_name": course.display_name_with_default,
+                'checkpoint_name': checkpoint_name,
+                "error": True,
+                "errorMsg": "No checkpoint found",
+                'platform_name': settings.PLATFORM_NAME,
+            }
+            return render_to_response("verify_student/incourse_reverify.html", context)
+        init_verification = SoftwareSecurePhotoVerification.get_initial_verification(user)
+        if not init_verification:
+            log.error("Could not submit verification attempt for user %s", request.user.id)
+            return redirect(reverse('verify_student_verify_later', kwargs={'course_id': unicode(course_key)}))
+
+        try:
+            b64_face_image = request.POST['face_image'].split(",")[1]
+            attempt = SoftwareSecurePhotoVerification(user=request.user)
+            attempt.upload_face_image(b64_face_image.decode('base64'))
+            attempt.photo_id_key = init_verification.photo_id_key
+            attempt.mark_ready()
+
+            attempt.save()
+            attempt.submit()
+
+            checkpoint.add_verification_attempt(attempt)
+            VerificationStatus.add_verification_status(checkpoint, user, "submitted")
+
+            # emit the reverification event
+            _track_reverification_events(
+                EVENT_NAME_USER_SUBMITTED_INCOURSE_REVERIFY, user.id, course_id, checkpoint_name
+            )
+
+            return HttpResponse()
+        except IndexError:
+            log.exception("Invalid image data during photo verification.")
+            return HttpResponseBadRequest(_("Invalid image data during photo verification."))
+        except:  # pylint: disable=broad-except
+            log.error("Could not submit verification attempt for user %s", request.user.id)
+            msg = _("Could not submit photos")
+            return HttpResponseBadRequest(msg)
+
+def _track_reverification_events(event_name, user_id, course_id, checkpoint):  # pylint: disable=invalid-name
+    """Track re-verification events for user against course checkpoints
+
+    Arguments:
+        user_id (str): The ID of the user generting the certificate.
+        course_id (unicode):  id associated with the course
+        checkpoint (str):  checkpoint name
+    Returns:
+        None
+
+    """
+    if settings.FEATURES.get('SEGMENT_IO_LMS') and hasattr(settings, 'SEGMENT_IO_LMS_KEY'):
+        tracking_context = tracker.get_tracker().resolve_context()
+        analytics.track(
+            user_id,
+            event_name,
+            {
+                'category': "verification",
+                'label': unicode(course_id),
+                'checkpoint': checkpoint
```

Comment:
Is marketing aware that this `checkpoint` value won't appear in Google Analytics? What you're doing here is fine, but if marketing is expecting to see this information in GA, they'll be disappointed.

This also seems like information that could be of interest to the research community; has anyone from the analytics team been consulted to see if this should be sent to the tracking logs?